src/url.rs

```diff
@@ -0,0 +1,75 @@
+use crate::{
+    bindings::*,
+    helpers::{checked_raw_str, checked_raw_string},
+};
+use std::marker::PhantomData;
+
+pub struct Url<'a, T: 'a> {
+    bufp: TSMBuffer,
+    offset: TSMLoc,
+    _phantom_data: PhantomData<&'a T>,
+}
+
+impl<'a, T> Url<'a, T> {
```

Comment:
In rust it's customary to name getters by nouns, so `.get_host()` would become `.host()`, `.get_scheme()` would become `.scheme()` and so on.